### PR TITLE
Remove Salt icon color and style overrides

### PR DESCRIPTION
**Pull request by lilyvc**
This PR removes the overrides for Salt Icon color and size. 

The color override is stopping the color prop from working on the Salt site, (added here: https://github.com/jpmorganchase/salt-ds/pull/2462). The Salt icon will now use currentColor by default, so the override shouldn't be needed anymore.

The Mosaic Icon now uses number for size to align with Salt.

### CODE REVIEW

File: packages/components/src/Icon/index.tsx [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 
-import styles from './styles.css';
-
 export interface IconProps {
   /** Additional class name for root class override. */
   className?: string;
   /** Name of the icon */
   name: IconNames;
   /** Size of Icon */
-  size?: 'small' | 'medium' | 'large';
```
Comment (joshwooding) [inline packages/components/src/Icon/index.tsx:12]: Do teams use Icon directly? If so will this cause adoption issues? It it worth supporting both and then mapping to the new values?

Comment (lilyvc) [reply packages/components/src/Icon/index.tsx:12]: I had a look in developer docs and couldn't find any uses of size

Comment (lilyvc) [reply packages/components/src/Icon/index.tsx:12]: I couldn't find any uses of it in the main docs repo

Comment (lilyvc) [reply packages/components/src/Icon/index.tsx:12]: It would be safer to keep this but I don't know if its necessary

Comment (DavieReid) [reply packages/components/src/Icon/index.tsx:12]: I think we should keep it for now.

**Review by DavieReid — APPROVED** [review]

**Pull request merged**